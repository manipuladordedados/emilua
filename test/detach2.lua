-- This test ensures that fibers whose joinable fiber handles got collected
-- will have their stacktrace printed to stderr (if err'ed)

local println = require('println')

collectgarbage("stop")

spawn(function()
    println('secondary fiber starts')
    error('tag', 0)
end)

println('secondary fiber spawned')
this_fiber.yield()
println('about to gc')
collectgarbage("collect")
println('end of main fiber')
