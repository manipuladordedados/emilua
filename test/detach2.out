secondary fiber spawned
secondary fiber starts
about to gc
Fiber 0x1 panicked: 'tag'
stack traceback:
	input.lua:10: in function <input.lua:8>
end of main fiber
